paddle/fluid/pybind/auto_parallel_py.cc

```diff
@@ -364,7 +364,30 @@ void BindAutoParallel(py::module *m) {
           py::arg("memo"))
       .def("__str__", &DeviceMesh::to_string);
 
-  py::enum_<phi::ReduceType>(*m, "ReduceType")
+  py::enum_<phi::ReduceType>(*m, "ReduceType", R"DOC(
+    Specify the type of operation used for paddle.distributed.Partial().
+    It should be one of the following values:
+
+        ReduceType.kRedSum
+        ReduceType.kRedMax
+        ReduceType.kRedMin
+        ReduceType.kRedProd
+        ReduceType.kRedAvg
+        ReduceType.kRedAny
+        ReduceType.kRedAll
+
+    Examples:
+        .. code-block:: python
+
+        >>> import paddle
+        >>> import paddle.distributed as dist
+        >>> mesh = dist.ProcessMesh([0, 1], dim_names=["x"])
+        >>> a = paddle.ones([10, 20])
+        >>> # doctest: +REQUIRES(env:DISTRIBUTED)
+        >>> # distributed tensor
+        >>> d_tensor = dist.shard_tensor(a, mesh, [dist.Partial(dist.ReduceType.kRedSum)])
```

Comment:
`Examples`、`.. code-block:: python` 和`代码片段`层层都得保持缩进，参考 [API文档英文模板](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/dev_guides/api_contributing_guides/api_docs_guidelines_cn.html#yingwenmuban) ，不然网页渲染会不正常。
```suggestion
    Examples:
        .. code-block:: python

            >>> import paddle
            >>> import paddle.distributed as dist
            >>> mesh = dist.ProcessMesh([0, 1], dim_names=["x"])
            >>> a = paddle.ones([10, 20])
            >>> # doctest: +REQUIRES(env:DISTRIBUTED)
            >>> # distributed tensor
            >>> d_tensor = dist.shard_tensor(a, mesh, [dist.Partial(dist.ReduceType.kRedSum)])
```

Reply:
DONE